patterns/choreography-to-orchestration.md

```diff
@@ -0,0 +1,59 @@
+# Convert Choreography to Orchestration
+
+![](images/choreography-to-orchestration/choreography.png)
+
+***
+
+![](images/choreography-to-orchestration/orchestration.png)
+
+***
+
+## Description:
+
+Choreography and Orchestration are two modes on how distributed microservices services can communicate with each another.
+
+In Choreography, every service works independently and achieves communication without tight control. Events flow between services without any centralized coordination. Each microservice listens for events of it's interest and does its own thing, following the event-driven paradigm
+
+In orchestration, there is a central service (the ‘orchestrator’) that controls the interaction between services. It coordinates the interaction and order in which services are invoked like a control flow for the business logic. 
+
+Many applications use both choreography and orchestration for different use cases. Communication between bounded contexts is often how choreography is used most effectively, within a bounded context, you need to control the sequence of service integration, maintain state, and handle errors and retries. A standard guideline is to use orchestration within the bounded context of a microservice, but use choreography between bounded-contexts. 
+
+The following article presents a comprehensive illustration of two interaction patterns, highlights their pros and cons, and gives the advice how to select or combine them within a given context.
+
+[Choreography vs Orchestration in the land of serverless](https://theburningmonk.com/2020/08/choreography-vs-orchestration-in-the-land-of-serverless/)
+
+
+## Solution:
+
+AWS Step Functions is a serverless orchestration service. A workflow in Step Functions is called a state machine, which simplifies implementation of common tasks and supports features like retries, handling failures, parallelization, service integrations, and observability. 
+
+As an example to convert Choreography to Orchestration by using AWS Step Function, here is the CDK implementation written in Typescript: [/implementation/choreography-to-orchestration](/implementation/choreography-to-orchestration)
+
+## Advantages:
+
+- Easier to implement as one system controls the flow between components. The work flow is modelled and source controlled
+
+- Simplifies end-to-end monitoring, reporting using built in visualization and audit histories.
+ 
+- Transaction management and Error management - easier to handle transaction errors, timeout, retries etc. 
+
+- Centralized business logic - everything can be seen in one place. No need for architecture diagrams and supporting documents. 
+
+
+## Limitations: 
+
+- Additional costs when using the Orchestration service.
```

Comment:
Can you provide data around it. Examples: What caused cost to increase, how much was the cost increase?